### DISCUSSION TITLE
Notification on job failure in pg_timetable

### DISCUSSION THREAD

**Discussion by ashx1901** [cybertec-postgresql/pg_timetable#487] [Q&A]
Hi, 

As per documentation, there is a way to notify on job completion/success. Could you please give me some examples where i can modify my existing job to notify my team when it fails.

Thanks

**Comment by pashagolub** [comment]
Helllo,

would you please quote the manual so I can be sure what part we are talking about?

Thanks in advance!

**Reply by ashx1901** [reply]
Sorry, I have the following queries if you can help me. I couldn't find answers of my below query in the documentation

1. In a multistep job how can i get notification if one of the steps fails.
2. How to create multistep conditional chain in pg_timetable

 I would be grateful to you if you can help me with some examples.
 
 Thanks

**Reply by pashagolub** [reply]
Here is the [example ](https://github.com/cybertec-postgresql/pg_timetable/blob/master/samples/ManyTasks.sql) showing how the task can get the fail/success information from `timetable.exeution_log` table. You can use that approach as well.

So in other words you don't get a notification but rather a task in your chain checks previous result codes.

Under the conditional chain you mean a simple IF statement or you want like to have completely different workflows depending on some condition? If the first, I would check that right in the SQL task. If the other you can have a task which will run another chain depending on the condition with `timetable.notify_chain_start()` function

**Reply by seanzechman** [reply]
It is not clear from example - how could one send an email if and only if the previous task failed?

**Reply by pashagolub** [reply]
As a quick answer here what I'm thinking about:

```mermaid
flowchart TD
    Task1[Task1: ignore_error=true] --> Task2(Task2: Adding a row to timetable.parameter if return code of the previous task > 0)
    Task2 --> ETask(Email: Sends if there is any row, just ignores otherwise)
```

In other words you execute every task in a chain no matter what. The difference is whether anything to send for an email task.
The hardest part is to check the return code. I hope [this example](https://pg-timetable.readthedocs.io/en/master/samples.html#access-previous-task-result-code-and-output-from-the-next-task) may throw some light on it. Feel free to ask additional question if case of difficulties.
